Assets/MixedRealityToolkit.Tests/PlayModeTests/BoundingBoxTests.cs

```diff
@@ -0,0 +1,282 @@
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Linq;
+using System;
+using UnityEditor;
+using System.Collections.Generic;
+using Assert = UnityEngine.Assertions.Assert;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    public class BoundingBoxTests
+    {
+        #region Utilities
+        [TearDown]
+        public void ShutdownMrtk()
+        {
+            TestUtilities.ShutdownMixedRealityToolkit();
+        }
+
+        /// <summary>
+        /// Instantiates a bounding box at 0, 0, -1.5f
+        /// box is at scale 1,1,1
+        /// </summary>
+        /// <returns></returns>
+        private BoundingBox InstantiateSceneAndDefaultBbox()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes(true);
+            TestUtilities.InitializePlayspace();
+
+            RenderSettings.skybox = null;
+
+            var cube = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            cube.transform.position = Vector3.forward * 1.5f;
+            BoundingBox bbox = cube.AddComponent<BoundingBox>();
+
+            MixedRealityPlayspace.PerformTransformation(
+            p =>
+            {
+                p.position = Vector3.zero;
+                p.LookAt(cube.transform.position);
+            });
+
+            bbox.transform.localScale *= 0.5f;
+            bbox.Active = true;
+
+            return bbox;
+        }
+        #endregion
+
+        [UnityTest]
+        public IEnumerator BBoxInstantiate()
+        {
+            var bbox = InstantiateSceneAndDefaultBbox();
+            yield return null;
+            Assert.IsNotNull(bbox);
+
+            GameObject.Destroy(bbox.gameObject);
+            // Wait for a frame to give Unity a change to actually destroy the object
+            yield return null;
+        }
+
+        /// <summary>
+        /// Test that if we update the bounds of a box collider, that the corners will move correctly
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator BBoxOverride()
+        {
+            var bbox = InstantiateSceneAndDefaultBbox();
+            yield return null;
+            bbox.BoundingBoxActivation = BoundingBox.BoundingBoxActivationType.ActivateOnStart;
+            bbox.HideElementsInInspector = false;
+            yield return null;
+
+            var newObject = new GameObject();
+            var bc = newObject.AddComponent<BoxCollider>();
+            bc.center = new Vector3(.25f, 0, 0);
+            bc.size = new Vector3(0.162f, 0.1f, 1);
+            bbox.BoundsOverride = bc;
+            yield return null;
+
+            // Get the dimensions of the corners
+            List<GameObject> corners = FindDescendantsContainingName(bbox.gameObject, "corner_");
```

Comment:
Can we add a function to BoundingBox to get the GameObject for all or one specific corner(s)? Could just be a wrapper around the FindDescendants, but would keep the logic inside BoundingBox.cs.

Reply:
ah great suggestion, thanks.